src/main/java/io/personium/core/PersoniumUnitConfig.java

```diff
@@ -533,60 +531,29 @@ protected PersoniumUnitConfig() {
-        Properties sysProps = System.getProperties();
-        //When reading succeeds, replace with member variable
-        if (!properties.isEmpty()) {
-            this.props.clear();
-            for (Map.Entry<Object, Object> entry : properties.entrySet()) {
-                if (!(entry.getKey() instanceof String)) {
-                    continue;
-                }
-                this.props.setProperty((String) entry.getKey(), (String) entry.getValue());
-            }
-        }
-        if (!propertiesOverride.isEmpty()) {
-            this.propsOverride.clear();
-            for (Map.Entry<Object, Object> entry : propertiesOverride.entrySet()) {
-                if (!(entry.getKey() instanceof String)) {
-                    continue;
-                }
-                this.propsOverride.setProperty((String) entry.getKey(), (String) entry.getValue());
-            }
-        }
         Map<String, String> env = System.getenv();
-        for (String key : env.keySet()) {
-            if (!key.startsWith("io.personium.")) {
-                continue;
-            }
-            String value = env.get(key);
-            if (value == null) {
-                continue;
-            }
-            log.info("From Env Vars, overriding config : " + key + "=" + value);
-            this.props.setProperty(key, value);
-        }
+        Properties sysProps = System.getProperties();
 
-        for (Object keyObj : propsOverride.keySet()) {
-            String key = (String) keyObj;
-            String value = this.propsOverride.getProperty(key);
-            if (value == null) {
-                continue;
-            }
-            log.info("From config file, overriding config : " + key + "=" + value);
-            this.props.setProperty(key, value);
-        }
-        for (Object keyObj : sysProps.keySet()) {
-            String key = (String) keyObj;
-            if (!key.startsWith("io.personium.")) {
-                continue;
-            }
-            String value = sysProps.getProperty(key);
-            if (value == null) {
-                continue;
+        this.props.clear();
+        overrideConf(properties, "default");
+        overrideConf(propertiesOverride, "config file");
+        overrideConf(env, "env vars");
+        overrideConf(sysProps, "system props");
+    }
+
+    private void overrideConf(Map<?, ?> propsOverride, String from) {
+        Map<String, String> override = propsOverride.entrySet().stream()
+            .filter(e -> e.getKey() instanceof String && e.getValue() instanceof String)
+            .filter(e -> ((String) e.getKey()).startsWith("io.personium."))
+            .filter(e -> e.getValue() != null)
+            .collect(Collectors.toMap(e -> (String) e.getKey(), e -> (String) e.getValue()));
+
+        override.forEach((key, value) -> {
+            if (!from.equals("default")) {
+                log.debug("From " + from + ", overriding config : " + key + "=" + value);
```

Comment:
Important! log.info write out unmasked server secret to log. It's better to set log level debug or mask secret information.

Reply:
Hmm.. I would prefer masking rather than making it to debug level.
I fear a bit about the impact of this change since ordinary users who do not change the log level to debug will now become unable to know what conf is used when booting the unit.

Masking should be necessary for very limited keys. 

- io.personium.core.security.secret16
- io.personium.core.security.auth.password.salt
- io.personium.core.masterToken